### PR TITLE
Add the effective license to the web app reporter

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Please see the commit messages for details.

### CODE REVIEW

**Review by tsteenbe — APPROVED** [review]
Did a quick review of JavaScript code, all looks good to me.

File: reporter/src/main/kotlin/reporters/evaluatedmodel/EvaluatedModelMapper.kt [hunk 1]
```diff
@@ -306,6 +307,11 @@ internal class EvaluatedModelMapper(private val input: ReporterInput) {
             detectedLicenses = detectedLicenses,
             detectedExcludedLicenses = detectedExcludedLicenses,
             concludedLicense = pkg.concludedLicense,
+            effectiveLicense = input.licenseInfoResolver.resolveLicenseInfo(pkg.id).filterExcluded().effectiveLicense(
+                LicenseView.CONCLUDED_OR_DECLARED_AND_DETECTED,
```
Comment (sschuberth) [inline reporter/src/main/kotlin/reporters/evaluatedmodel/EvaluatedModelMapper.kt:311]: BTW, I just checked all callers of `effectiveLicense()`, and we currently sometimes use `LicenseView.CONCLUDED_OR_DECLARED_AND_DETECTED` (like here), but sometimes `LicenseView.ALL`. Would you mind double-checking whether that makes sense?

Comment (fviernau) [inline reporter/src/main/kotlin/reporters/evaluatedmodel/EvaluatedModelMapper.kt:311]: Have license choices been already applied and if so, does the view need to be consistent with what was supplied to `applyChoices()` as view?

Comment (mnonnenmacher) [reply reporter/src/main/kotlin/reporters/evaluatedmodel/EvaluatedModelMapper.kt:311]: We only use `LicenseView.ALL` in the `DownloaderCommand` which is a special case I believe, and in tests. `LicenseView.CONCLUDED_OR_DECLARED_AND_DETECTED` is what we consider the default and use in several other places, especially the static HTML reporter. We might want to consider making the default configurable later on.

Comment (mnonnenmacher) [reply reporter/src/main/kotlin/reporters/evaluatedmodel/EvaluatedModelMapper.kt:311]: Yes, the choices are applied inside the `effectiveLicense` function based on the supplied view.

**Comment by mnonnenmacher** [comment]
> Did a quick review of JavaScript code, all looks good to me.

I have implemented it exactly the same as the concluded license and have manually verified that it works.

**Pull request merged**